internal/commands/app/new_app_input_test.go

```diff
@@ -0,0 +1,110 @@
+package app
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/10gen/realm-cli/internal/cloud/atlas"
+	"github.com/10gen/realm-cli/internal/cloud/realm"
+	"github.com/10gen/realm-cli/internal/utils/test/assert"
+	"github.com/10gen/realm-cli/internal/utils/test/mock"
+
+	"github.com/Netflix/go-expect"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+)
+
+func TestAppNewAppInputsResolveFrom(t *testing.T) {
+	t.Run("should do nothing if from is not set", func(t *testing.T) {
+		var i newAppInputs
+		f, err := i.resolveFrom(nil, nil, nil)
+		assert.Nil(t, err)
+		assert.Equal(t, from{}, f)
+	})
+
+	testApp := realm.App{
+		ID:          primitive.NewObjectID().Hex(),
+		GroupID:     primitive.NewObjectID().Hex(),
+		ClientAppID: "test-app-abcde",
+		Name:        "test-app",
+	}
+
+	for _, tc := range []struct {
+		description    string
+		inputs         newAppInputs
+		procedure      func(c *expect.Console)
+		findAppErr     error
+		groupsErr      error
+		expectedFrom   from
+		expectedFilter realm.AppFilter
+		expectedErr    error
+	}{
+		{
+			description:    "should return the app id and group id of specified app when project and from are set",
```

Comment:
let's reorganize this test file a bit... assuming we make the change to have two separate functions for resolving `Project` and resolving `From`, that'll be the first divide of test cases.  after that, within those groups lets have tests (or test tables if need be) for the ones that are interactive and need a pseudo terminal and then ones that aren't interactive and can roll with a `nil`/simple ui